test/e2e/serial/tests/configuration.go

```diff
@@ -693,9 +697,215 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+
+			cpuNum, ok := cpuQty.AsInt64()
+			Expect(ok).To(BeTrue(), "invalid CPU resource in zone %q node %q: %v", referenceZone.Name, referenceNode.Name, cpuQty)
+			klog.Infof("available cpus per numa on the node %d, ", cpuNum)
+
+			cpuResources := strconv.Itoa(int(float64(cpuNum) * cpuResourcePercentage))
+			klog.Infof("cpu resources requested in the deployment to create TAE %s", cpuResources)
+
+			By("fetching the matching kubeletconfig for the numaresources-operator cr")
+			var nroOperObj nropv1.NUMAResourcesOperator
+			nroKey := objects.NROObjectKey()
+			err := fxt.Client.Get(ctx, nroKey, &nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroKey.String())
+
+			mcps, err := nropmcp.GetListByNodeGroupsV1(ctx, fxt.Client, nroOperObj.Spec.NodeGroups)
+			Expect(err).ToNot(HaveOccurred(), "cannot get MCPs associated with NUMAResourcesOperator %q", nroOperObj.Name)
+
+			kcList := &machineconfigv1.KubeletConfigList{}
+			err = fxt.Client.List(ctx, kcList)
+			Expect(err).ToNot(HaveOccurred())
+
+			var targetedKC *machineconfigv1.KubeletConfig
+			for _, mcp := range mcps {
+				for i := 0; i < len(kcList.Items); i++ {
+					kc := &kcList.Items[i]
+					kcMcpSel, err := metav1.LabelSelectorAsSelector(kc.Spec.MachineConfigPoolSelector)
+					Expect(err).ToNot(HaveOccurred())
+
+					if kcMcpSel.Matches(labels.Set(mcp.Labels)) {
+						targetedKC = kc
+					}
+				}
+			}
+			Expect(targetedKC).ToNot(BeNil(), "there should be at least one kubeletconfig.machineconfiguration object")
+
+			kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+			Expect(err).ToNot(HaveOccurred())
+			initialTopologyManagerPolicy := kcObj.TopologyManagerPolicy
+
+			By("modifying Topology Manager Policy under kubeletconfig to none")
+			Eventually(func(g Gomega) {
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Before restarting the scheduler, we change the topology manager policy to 'none'.
+				// This ensures that if we later change it to 'single NUMA node' and attempt to create a deployment that should fail on TAE,
+				// the deployment will fail as expected. If the policy were still set to 'none', it would allow the deployment to proceed.
+				kcObj.TopologyManagerPolicy = v1beta1.NoneTopologyManagerPolicy
+				err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = fxt.Client.Update(ctx, targetedKC)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+			// TODO find a solution to wait for mcp to start updating only if the tm policy
+			// has changed otherwise it will be stuck here
+			waitForMcpUpdate(fxt.Client, ctx, mcps)
```

Comment:
you can verify that even before by checking if `initialTopologyManagerPolicy != NoneTopologyManagerPolicy` is true only then execute the eventually and wait for the update to happen

Reply:
right thanks!